### DISCUSSION TITLE
How to make a recipe depend on image being built

### DISCUSSION THREAD

**Discussion by tirankas** [OE4T/meta-tegra#750] [General]
Hi Guys,
My case is that a specific recipe should do something based on the image being built.
I can duplicate the recipe and select each of them from the image via image_install, however this seems overkill and ugly.

Is there a way I could perform a specific function in the recipe based on the image, something like do_install_append_myimage?

Thanks

**Comment by madisongh** [comment]
An override like that isn't something that the build system supports, since an image is essentially just a list of packages to be installed, and the intent is to maximize the sharing of packages between different images you might build.

If you just need to install a different set of files built from your recipe into the different images, then you can do this from a single recipe, and use `PACKAGES` and `FILES_xxx` to put the files into different packages.  Then you'd include the appropriate packages in each image. For an example, see the [tegra-boot-tools](https://github.com/OE4T/meta-tegra/blob/master/recipes-bsp/tools/tegra-boot-tools_2.4.0.bb) recipe here: it breaks things up into multiple packages, so the appropriate subset of the built programs can be included in an initramfs image (the `-earlyboot` package) vs. a regular rootfs image (the `-lateboot` package and others as needed).

If, on the other hand, you need to build the recipe differently based on the image (e.g., with different compile-time options), then you'd need to create two separate recipes. I can't think of a good example of this to point to, since it's not considered good practice (for open-source packages, at least). I have had to do this in some projects in the past, though.
